crates/sui-types/src/messages.rs

```diff
@@ -1363,8 +1363,9 @@ pub enum ExecutionFailureStatus {
 #[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize, Hash)]
 pub struct MoveLocation {
     pub module: ModuleId,
-    pub function: u16,
+    pub function_idx: u16,
     pub instruction: CodeOffset,
+    pub message: Option<String>,
```

Comment:
Unfortunately, I don't think we want this string here. When we include a string in errors like this, it bakes in the string and all the formatting and everything into the protocol, which can make things a bit brittle. 

I'd be okay with including/resolving the string for the function name and including that in the `MoveLocation`. Thoughts?

Reply:
Makes sense! What comes to mind then is to parse out the function name from `error.message`? Still brittle but at least we control/gate/process the values we care about this way.

I don't see an obvious way to resolve the function name from other data structures in this area of the code (Sui side), unless you had a better proposal there? It seems really involved to propagate out via `VMError` in move-lang repo.